content/articles/building-a-react-app-with-typescript/index.md

```diff
@@ -0,0 +1,276 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /building-a-react-app-with-typescript/
+title: Building A React App With Typescript
+description: This article will give the reader a guide on how to build a React app with TypeScript.
+author: babatunde-koiki
+date: 2021-08-18T11:44-06
+topics: [React, Typescript]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/building-a-react-app-with-typescript/react-ts.png
+    alt: React App Using TypeScript
+---
+
+## Introduction
+
+![](react-ts.png)
+
+React is one of the most popular and best front-end libraries as of today this has made a lot of developers make it their primary library used for creating front-end applications. Typescript on the other hand is a strict and statically typed programming language which makes our javascript code more predictable. As developers, our goal is not just to create products but to create products that we can always predict their behavior, this is one of the main reasons why Typescript was created. Using Typescript to create our React applications will make our react applications more predictable as we will be able to catch alot of errors at runtime(during compilation).
+In this article, I'd explain the following while building a simple Todo Application:
+1. How to bootstrap a react typescript application
+2. What are types and interfaces in typescript and how to use them
+3. How to declare types for states and props in functional components
+4. Events and how to deal with event types in our react applications
+5. How to declare types for states and props in class base components.
+
+If you are interested in the code alone, you can check out this Github repo [here](https://github.com/Babatunde13/react-ts-section)
+
+This article isn't a beginner guide to React or Typescript but instead a guide on how to integrate Typescript in React applications.
+
+
+### Setup a React Typescript Application
+We use Create React App to set up our react application that way we do not need to worry about all the webpack and babel stress, as you might have guessed to set up a react typescript application we'd still use create-react-app just with some extra flags. In our terminal we'd type the following to bootstrap our application
+
+```bash
+npx create-react-app todo-app --template typescript
+```
+
+This command will create a react typescript application for us then we can set up the server as we will normally. Our app structure should look like this
+
+![](./filestructure.png)
+
```

Comment:
Hey you should use `/section-engineering-education/building-a-react-app-with-typescript/imageName` as the path. Kindly change every instance. This is your first article and therefore make sure to go through the [new contributors guide](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/UPLOAD_INSTRUCTIONS.md)

Reply:
👍